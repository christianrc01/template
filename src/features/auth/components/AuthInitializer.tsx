import { useEffect, useState } from "react";
import { useMsal } from "@azure/msal-react";
import { InteractionStatus } from "@azure/msal-browser";
import LoadingSpinner from "@/shared/components/common/LoadingSpinner";
import { RouterProvider, createBrowserRouter } from "react-router-dom";
import { routes } from "@/app/routing/routes";

const router = createBrowserRouter(routes);

function AuthInitializer() {
  const { instance, accounts, inProgress } = useMsal();
  const [initialized, setInitialized] = useState(false);

  useEffect(() => {
    if (inProgress === InteractionStatus.None) {
      setInitialized(true);
    }
  }, [inProgress]);

  useEffect(() => {
    if (!initialized) return;

    if (accounts.length === 0 && inProgress === InteractionStatus.None) {
      console.log("🔐 No session. Redirecting to Azure login...");
      instance
        .loginRedirect({
          scopes: ["openid", "profile", "User.Read"],
        })
        .catch(console.error);
    } else if (accounts.length > 0) {
      console.log("✅ Session active:", accounts[0].username);
    }
  }, [initialized, accounts, inProgress, instance]);

  if (inProgress !== InteractionStatus.None || !initialized) {
    return (
      <main
        className="grid place-items-center w-screen"
        role="status"
        aria-live="polite"
      >
        <LoadingSpinner />
      </main>
    );
  }

  return <RouterProvider router={router} />;
}

export default AuthInitializer;
