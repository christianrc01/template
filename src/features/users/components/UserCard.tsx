import IconLink from "../../../shared/icons/IconLink";
import IconMail from "../../../shared/icons/IconMail";
import IconPhone from "../../../shared/icons/IconPhone";
import type { UserCardProps } from "../../../shared/types/IUser";

function UserCard({ user }: UserCardProps) {
  return (
    <div className="p-6 border border-gray-200 rounded-lg shadow-sm hover:shadow-md transition-shadow bg-white max-w-md">
      <div className="flex items-start space-x-4">
        <div className="flex-1">
          <h3 className="text-xl font-bold text-gray-800">{user.name}</h3>
          <p className="text-gray-500">@{user.username}</p>

          <div className="mt-4 space-y-2">
            <p className="text-gray-600 flex items-center gap-2">
              <IconMail />
              {user.email}
            </p>

            <p className="text-gray-600 flex items-center gap-2">
              <IconPhone />
              {user.phone}
            </p>

            <p className="text-gray-600 flex items-center gap-2">
              <IconLink />
              <a
                href={`https://${user.website}`}
                target="_blank"
                rel="noopener noreferrer"
                className="hover:text-blue-500 hover:underline"
              >
                {user.website}
              </a>
            </p>
          </div>

          <div className="mt-4 p-3 bg-gray-50 rounded-lg">
            <h4 className="font-medium text-gray-700">Address</h4>
            <p className="text-gray-600">
              {user.address.street}, {user.address.suite}
              <br />
              {user.address.city}, {user.address.zipcode}
            </p>
            <p className="text-sm text-gray-500 mt-1">
              Coordinates: {user.address.geo.lat}, {user.address.geo.lng}
            </p>
          </div>

          <div className="mt-3 p-3 bg-gray-50 rounded-lg">
            <h4 className="font-medium text-gray-700">Company</h4>
            <p className="text-gray-800 font-medium">{user.company.name}</p>
            <p className="text-gray-600 italic">"{user.company.catchPhrase}"</p>
            <p className="text-sm text-gray-500 mt-1">{user.company.bs}</p>
          </div>
        </div>
      </div>
    </div>
  );
}

export default UserCard;
