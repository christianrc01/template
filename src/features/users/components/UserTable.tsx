import {
  Grid,
  GridColumn,
  GridToolbar,
  type GridDataStateChangeEvent,
} from "@progress/kendo-react-grid";
import { process, type State } from "@progress/kendo-data-query";
import { useState } from "react";
import type { UserProps } from "@/shared/interfaces/IUser";
import "@/shared/styles/kendo-overrides.css";

function UserTable({ users }: { users: UserProps["user"][] }) {
  const [dataState, setDataState] = useState<State>({
    skip: 0,
    take: 10,
    sort: [{ field: "name", dir: "asc" }],
  });

  const handleDataStateChange = (e: GridDataStateChangeEvent) => {
    setDataState((prev) => ({
      ...prev,
      ...e.dataState,
      skip: e.dataState.skip ?? prev.skip,
      take: e.dataState.take ?? prev.take,
      sort: e.dataState.sort ?? prev.sort,
    }));
  };

  const processedUsers = users.map((user) => ({
    ...user,
    website: `https://${user.website}`,
  }));

  const processedData = process(processedUsers, dataState);
  const commonColumnProps = {
    headerClassName:
      "font-bold !text-gray-700 dark:!text-gray-300 !bg-gray-100 dark:!bg-gray-700 px-4 py-3 !border-b !border-gray-200 dark:!border-gray-600",
    className:
      "px-4 py-3 !text-gray-800 dark:!text-gray-200 !border-b !border-gray-100 dark:!border-gray-600 !bg-white/80 dark:!bg-gray-800/90",
  };

  const columns = [
    {
      field: "name",
      title: "Name",
      ...commonColumnProps,
    },
    {
      field: "username",
      title: "Username",
      ...commonColumnProps,
    },
    {
      field: "email",
      title: "Email",
      ...commonColumnProps,
    },
    {
      field: "phone",
      title: "Phone",
      ...commonColumnProps,
    },
    {
      field: "website",
      title: "Website",
      ...commonColumnProps,
    },
    {
      field: "address.city",
      title: "City",
      ...commonColumnProps,
    },
    {
      field: "company.name",
      title: "Company",
      ...commonColumnProps,
    },
  ];

  return (
    <div className="p-6">
      <Grid
        data={processedData}
        pageable={true}
        sortable={true}
        {...dataState}
        onDataStateChange={handleDataStateChange}
        className="rounded-2xl shadow-sm !border !border-gray-200 dark:!border-gray-600 !bg-white dark:!bg-gray-800"
      >
        <GridToolbar className="rounded-t-2xl !bg-white dark:!bg-gray-800">
          <div className="flex justify-between items-center w-full p-4 !bg-gradient-to-r !from-blue-50 !to-gray-50 dark:!from-gray-700 dark:!to-gray-800 !border-b !border-gray-200 dark:!border-gray-600 rounded-t-lg">
            <h2 className="text-xl font-bold !text-gray-800 dark:!text-gray-200">
              Users List
            </h2>
            <div className="text-sm !text-gray-600 dark:!text-gray-300 !bg-white dark:!bg-gray-700 px-3 py-1 rounded-full shadow-sm">
              Total: <span className="font-semibold">{users.length}</span> users
            </div>
          </div>
        </GridToolbar>

        {columns.map((column) => (
          <GridColumn key={column.field} {...column} />
        ))}
      </Grid>
    </div>
  );
}

export default UserTable;
