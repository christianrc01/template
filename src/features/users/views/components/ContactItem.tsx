function ContactItem({
  icon,
  text,
}: {
  icon: React.ReactNode;
  text: React.ReactNode;
}) {
  return (
    <div className="text-gray-600 dark:text-gray-400 flex items-center gap-2 min-w-0">
      {icon}
      <span className="break-words block text-sm sm:text-base leading-tight min-w-0">
        {text}
      </span>
    </div>
  );
}

export default ContactItem;
