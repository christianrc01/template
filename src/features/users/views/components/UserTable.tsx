import {
  Grid,
  GridColumn,
  GridToolbar,
  type GridColumnProps,
  type GridDataStateChangeEvent,
} from "@progress/kendo-react-grid";
import { process, type State } from "@progress/kendo-data-query";
import { useState } from "react";
import type { UserProps } from "@/features/users/types/IUser";
import TableLinkCell from "@/shared/views/components/common/TableLinkCell";
import { ROUTE_PATHS } from "@/app/routes";

function UserTable({ users }: { users: UserProps["user"][] }) {
  const [dataState, setDataState] = useState<State>({
    skip: 0,
    take: 10,
    sort: [{ field: "name", dir: "asc" }],
  });

  const handleDataStateChange = (e: GridDataStateChangeEvent) => {
    setDataState((prev) => ({
      ...prev,
      ...e.dataState,
      skip: e.dataState.skip ?? prev.skip,
      take: e.dataState.take ?? prev.take,
      sort: e.dataState.sort ?? prev.sort,
    }));
  };

  const processedData = process(users, dataState);
  const commonColumnProps: Partial<GridColumnProps> = {
    headerClassName:
      "font-bold !text-gray-700 dark:!text-gray-300 !bg-gray-100 dark:!bg-gray-700 px-4 py-3 !border-b !border-gray-200 dark:!border-gray-600",
    className:
      "px-4 py-3 !text-gray-800 dark:!text-gray-200 !border-b !border-gray-100 dark:!border-gray-600 !bg-white/80 dark:!bg-gray-800/90",
  };

  return (
    <section
      aria-labelledby="users-table-title"
      className="overflow-x-auto w-full"
    >
      <Grid
        data={processedData}
        pageable={true}
        sortable={true}
        {...dataState}
        onDataStateChange={handleDataStateChange}
        aria-label="User data table"
        className="min-w-[600px] sm:min-w-full rounded-2xl shadow-sm !border !border-gray-200 dark:!border-gray-600 !bg-white dark:!bg-gray-800"
      >
        <GridToolbar className="rounded-t-2xl !bg-white dark:!bg-gray-800">
          <div className="flex justify-between items-center w-full p-4 !bg-gradient-to-r !from-blue-50 !to-gray-50 dark:!from-gray-700 dark:!to-gray-800 !border-b !border-gray-200 dark:!border-gray-600 rounded-t-lg">
            <h2
              id="users-table-title"
              className="text-xl font-bold !text-gray-800 dark:!text-gray-200"
            >
              Users List
            </h2>
            <p
              className="text-sm !text-gray-600 dark:!text-gray-300 !bg-white dark:!bg-gray-700 px-3 py-1 rounded-full shadow-sm"
              aria-live="polite"
            >
              Total: <span className="font-semibold">{users.length}</span> users
            </p>
          </div>
        </GridToolbar>

        <GridColumn
          field="name"
          title="Name"
          {...commonColumnProps}
          cells={{
            data: (props) => (
              <TableLinkCell
                {...props}
                getHref={(user) =>
                  ROUTE_PATHS.USER.replace(":id", user.id.toString())
                }
                getText={(user) => user.name}
              />
            ),
          }}
        />

        <GridColumn field="username" title="Username" {...commonColumnProps} />
        <GridColumn field="email" title="Email" {...commonColumnProps} />
        <GridColumn field="phone" title="Phone" {...commonColumnProps} />
        <GridColumn
          field="address.street"
          title="Street"
          {...commonColumnProps}
        />
        <GridColumn field="address.city" title="City" {...commonColumnProps} />
        <GridColumn
          field="company.name"
          title="Company"
          {...commonColumnProps}
        />
      </Grid>
    </section>
  );
}

export default UserTable;
