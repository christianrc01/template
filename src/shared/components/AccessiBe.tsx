import { useEffect } from "react";

function AccessiBe() {
  useEffect(() => {
    if (import.meta.env.MODE !== "production") return;
    const script = document.createElement("script");
    script.src = "https://acsbapp.com/apps/app/dist/js/app.js";
    script.async = true;
    script.onload = () => {
      window.acsbJS?.init({
        language: "es",
        position: "right",
        triggerIcon: "people",
        triggerColor: "#146FF8",
        leadColor: "#146FF8",
        triggerSize: "bottom",
        triggerOffsetX: 20,
        triggerOffsetY: 20,
        triggerRadius: "50%",
      });
    };
    document.head.appendChild(script);
    return () => {
      document.head.removeChild(script);
    };
  }, []);

  return null;
}

export default AccessiBe;
