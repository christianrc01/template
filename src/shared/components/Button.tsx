interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant?: "primary" | "outline-white";
}

export const Button = ({
  variant = "primary",
  children,
  className = "",
  ...props
}: ButtonProps) => {
  const baseClasses = "px-4 py-2 rounded transition-colors font-medium";
  const variantClasses = {
    primary: "bg-blue-500 text-white hover:bg-blue-600",
    "outline-white": "border border-white hover:bg-white hover:text-blue-600",
  };

  return (
    <button
      className={`${baseClasses} ${variantClasses[variant]} ${className}`}
      {...props}
    >
      {children}
    </button>
  );
};
