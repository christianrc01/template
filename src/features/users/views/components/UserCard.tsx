import IconMail from "@/shared/icons/IconMail";
import IconPeople from "@/shared/icons/IconPeople";
import type { UserProps } from "@/features/users/types/IUser";
import ContactItem from "@/features/users/views/components/ContactItem";
import IconLink from "@/shared/icons/IconLink";
import Icon from "@/shared/views/components/common/Icon";

function UserCard({ user }: UserProps) {
  return (
    <article
      aria-labelledby={`user-${user.id}-name`}
      className="max-w-md w-full p-6 border border-gray-200 dark:border-gray-600 rounded-4xl shadow-sm hover:shadow-md transition-shadow bg-white dark:bg-gray-800"
    >
      <div className="flex flex-col sm:flex-row sm:items-start sm:space-x-4 space-y-4 sm:space-y-0">
        <div className="flex-1">
          <h3 id={`user-${user.id}-name`}>{user.name}</h3>
          <p className="text-gray-500 dark:text-gray-400">@{user.username}</p>

          <div className="mt-4 space-y-2">
            <ContactItem icon={<Icon icon={IconMail} />} text={user.email} />
            <ContactItem icon={<Icon icon={IconPeople} />} text={user.phone} />
            <ContactItem
              icon={<Icon icon={IconLink} />}
              text={
                <a
                  href={`https://${user.website}`}
                  draggable={false}
                  target="_blank"
                  rel="noopener noreferrer"
                  className="hover:underline"
                >
                  {user.website}
                </a>
              }
            />
          </div>

          <address className="mt-4 p-3 bg-gray-50 dark:bg-gray-700 rounded-2xl">
            <h4>Address</h4>
            <p className="text-gray-600 dark:text-gray-400">
              {user.address.street}, {user.address.suite}
              <br />
              {user.address.city}, {user.address.zipcode}
            </p>
            <p className="text-sm text-gray-500 dark:text-gray-500 mt-1">
              Coordinates: {user.address.geo.lat}, {user.address.geo.lng}
            </p>
          </address>

          <div className="mt-3 p-3 bg-gray-50 dark:bg-gray-700 rounded-2xl">
            <h4>Company</h4>
            <p className="text-gray-800 dark:text-gray-200 font-medium">
              {user.company.name}
            </p>
            <p className="text-gray-600 dark:text-gray-400 italic">
              "{user.company.catchPhrase}"
            </p>
            <p className="text-sm text-gray-500 dark:text-gray-500 mt-1">
              {user.company.bs}
            </p>
          </div>
        </div>
      </div>
    </article>
  );
}

export default UserCard;
