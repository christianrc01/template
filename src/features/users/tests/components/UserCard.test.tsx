import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import { UserCard } from "../../components/UserCard";
import { mockUser } from "../data/mockUsers";

describe("UserCard Component", () => {
  it("renders user basic information correctly", () => {
    render(<UserCard user={mockUser} />);

    expect(screen.getByText(mockUser.name)).toBeInTheDocument();
    expect(screen.getByText(`@${mockUser.username}`)).toBeInTheDocument();
    expect(screen.getByText(mockUser.email)).toBeInTheDocument();
    expect(screen.getByText(mockUser.phone)).toBeInTheDocument();
  });

  it("renders the website link correctly", () => {
    render(<UserCard user={mockUser} />);

    const websiteLink = screen.getByText(mockUser.website);
    expect(websiteLink).toBeInTheDocument();
    expect(websiteLink.closest("a")).toHaveAttribute(
      "href",
      `https://${mockUser.website}`
    );
    expect(websiteLink.closest("a")).toHaveAttribute("target", "_blank");
    expect(websiteLink.closest("a")).toHaveAttribute(
      "rel",
      "noopener noreferrer"
    );
  });

  it("renders address information correctly", () => {
    render(<UserCard user={mockUser} />);

    const addressSection = screen.getByText("Address").parentElement;
    expect(addressSection).toHaveTextContent(
      `${mockUser.address.street}, ${mockUser.address.suite}`
    );
    expect(addressSection).toHaveTextContent(
      `${mockUser.address.city}, ${mockUser.address.zipcode}`
    );
    expect(
      screen.getByText(
        `Coordinates: ${mockUser.address.geo.lat}, ${mockUser.address.geo.lng}`
      )
    ).toBeInTheDocument();
  });

  it("renders company information correctly", () => {
    render(<UserCard user={mockUser} />);

    expect(screen.getByText("Company")).toBeInTheDocument();
    expect(screen.getByText(mockUser.company.name)).toBeInTheDocument();
    expect(
      screen.getByText(`"${mockUser.company.catchPhrase}"`)
    ).toBeInTheDocument();
    expect(screen.getByText(mockUser.company.bs)).toBeInTheDocument();
  });

  it("applies correct styling classes", () => {
    const { container } = render(<UserCard user={mockUser} />);

    // Verify that the card has the correct classes
    expect(container.firstChild).toHaveClass(
      "p-6 border border-gray-200 rounded-lg shadow-sm hover:shadow-md"
    );

    // Verify that the address section has the correct classes
    expect(screen.getByText("Address").parentElement).toHaveClass(
      "mt-4 p-3 bg-gray-50 rounded-lg"
    );
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<UserCard user={mockUser} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
