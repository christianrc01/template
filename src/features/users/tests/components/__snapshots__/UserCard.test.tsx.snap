// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserCard Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="p-6 border border-gray-200 rounded-lg shadow-sm hover:shadow-md transition-shadow bg-white max-w-md"
  >
    <div
      class="flex items-start space-x-4"
    >
      <div
        class="flex-1"
      >
        <h3
          class="text-xl font-bold text-gray-800"
        >
          Leanne Graham
        </h3>
        <p
          class="text-gray-500"
        >
          @Bret
        </p>
        <div
          class="mt-4 space-y-2"
        >
          <p
            class="text-gray-600 flex items-center gap-2"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 20.5H7C4 20.5 2 19 2 15.5V8.5C2 5 4 3.5 7 3.5H17C20 3.5 22 5 22 8.5V15.5C22 19 20 20.5 17 20.5Z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-miterlimit="10"
                stroke-width="1.5"
              />
              <path
                d="M17 9L13.87 11.5C12.84 12.32 11.15 12.32 10.12 11.5L7 9"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-miterlimit="10"
                stroke-width="1.5"
              />
            </svg>
            Sincere@april.biz
          </p>
          <p
            class="text-gray-600 flex items-center gap-2"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21.97 18.33C21.97 18.69 21.89 19.06 21.72 19.42C21.55 19.78 21.33 20.12 21.04 20.44C20.55 20.98 20.01 21.37 19.4 21.62C18.8 21.87 18.15 22 17.45 22C16.43 22 15.34 21.76 14.19 21.27C13.04 20.78 11.89 20.12 10.75 19.29C9.6 18.45 8.51 17.52 7.47 16.49C6.44 15.45 5.51 14.36 4.68 13.22C3.86 12.08 3.2 10.94 2.72 9.81C2.24 8.67 2 7.58 2 6.54C2 5.86 2.12 5.21 2.36 4.61C2.6 4 2.98 3.44 3.51 2.94C4.15 2.31 4.85 2 5.59 2C5.87 2 6.15 2.06 6.4 2.18C6.66 2.3 6.89 2.48 7.07 2.74L9.39 6.01C9.57 6.26 9.7 6.49 9.79 6.71C9.88 6.92 9.93 7.13 9.93 7.32C9.93 7.56 9.86 7.8 9.72 8.03C9.59 8.26 9.4 8.5 9.16 8.74L8.4 9.53C8.29 9.64 8.24 9.77 8.24 9.93C8.24 10.01 8.25 10.08 8.27 10.16C8.3 10.24 8.33 10.3 8.35 10.36C8.53 10.69 8.84 11.12 9.28 11.64C9.73 12.16 10.21 12.69 10.73 13.22C11.27 13.75 11.79 14.24 12.32 14.69C12.84 15.13 13.27 15.43 13.61 15.61C13.66 15.63 13.72 15.66 13.79 15.69C13.87 15.72 13.95 15.73 14.04 15.73C14.21 15.73 14.34 15.67 14.45 15.56L15.21 14.81C15.46 14.56 15.7 14.37 15.93 14.25C16.16 14.11 16.39 14.04 16.64 14.04C16.83 14.04 17.03 14.08 17.25 14.17C17.47 14.26 17.7 14.39 17.95 14.56L21.26 16.91C21.52 17.09 21.7 17.3 21.81 17.55C21.91 17.8 21.97 18.05 21.97 18.33Z"
                stroke="currentColor"
                stroke-miterlimit="10"
                stroke-width="1.5"
              />
            </svg>
            1-770-736-8031 x56442
          </p>
          <p
            class="text-gray-600 flex items-center gap-2"
          >
            <svg
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3.27 12C2.48 11.05 2 9.83 2 8.5C2 5.48 4.47 3 7.5 3H12.5C15.52 3 18 5.48 18 8.5C18 11.52 15.53 14 12.5 14H10"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
              <path
                d="M20.73 12C21.52 12.95 22 14.17 22 15.5C22 18.52 19.53 21 16.5 21H11.5C8.48 21 6 18.52 6 15.5C6 12.48 8.47 10 11.5 10H14"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
            <a
              class="hover:text-blue-500 hover:underline"
              href="https://hildegard.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              hildegard.org
            </a>
          </p>
        </div>
        <div
          class="mt-4 p-3 bg-gray-50 rounded-lg"
        >
          <h4
            class="font-medium text-gray-700"
          >
            Address
          </h4>
          <p
            class="text-gray-600"
          >
            Kulas Light, Apt. 556
            <br />
            Gwenborough, 92998-3874
          </p>
          <p
            class="text-sm text-gray-500 mt-1"
          >
            Coordinates: -37.3159, 81.1496
          </p>
        </div>
        <div
          class="mt-3 p-3 bg-gray-50 rounded-lg"
        >
          <h4
            class="font-medium text-gray-700"
          >
            Company
          </h4>
          <p
            class="text-gray-800 font-medium"
          >
            Romaguera-Crona
          </p>
          <p
            class="text-gray-600 italic"
          >
            "Multi-layered client-server neural-net"
          </p>
          <p
            class="text-sm text-gray-500 mt-1"
          >
            harness real-time e-markets
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
